client/src/connection/itc/script.ts

```diff
@@ -224,5 +243,86 @@ class SASRunner{
 
     Write-Host "${LineCodes.ResultsFetchedCode}"
   }
+  
+  [void]GetDatasetRecords([string]$tableName, [int]$start = 0, [int]$limit = 100) {
+    $objRecordSet = New-Object -comobject ADODB.Recordset
+    $objRecordSet.ActiveConnection = $this.dataConnection # This is needed to set the properties for sas formats.
+    $objRecordSet.Properties.Item("SAS Formats").Value = "_ALL_"
+
+    $objRecordSet.Open(
+      $tableName, 
+      [System.Reflection.Missing]::Value, # Use the active connection
+      2,  # adOpenDynamic
+      1,  # adLockReadOnly
+      512 # adCmdTableDirect
+    )
+
+    $records = [List[List[object]]]::new()
+    $fields = $objRecordSet.Fields.Count
+
+    if ($objRecordSet.EOF) {
+      Write-Host '{"rows": [], "count": 0}'
+      return
+    }
+
+    $objRecordSet.AbsolutePosition = $start + 1
+
+    for ($j = 0; $j -lt $limit -and $objRecordSet.EOF -eq $False; $j++) {
+      $cell = [List[object]]::new()
+      for ($i = 0; $i -lt $fields; $i++) {
+        $cell.Add($objRecordSet.Fields.Item($i).Value)
+      }
+      $records.Add($cell)
+      $objRecordSet.MoveNext()
+    }
+    $objRecordSet.Close()
+
+    $objRecordSet.Open(
+      "SELECT COUNT(1) FROM $tableName", 
```

Comment:
Does it make sense to pass item name and library separately and request this information from sasHelp.vTables using `SELECT nobs FROM sasHelp.vTable WHERE libName = "$item.library" and memName = "$item.library"`
Although in both cases CPU time is extremely small, for 3 million records vTable approach is around 8 times faster and probably uses less resources.

Reply:
I realized it is better to keep the current approach. Later if we want to add filters in the data view, the current approach will be better suited.